common/src/outcome_fee.rs

```diff
@@ -112,6 +110,44 @@ where
         }
         self
     }
+
+    pub fn reduce(mut self, other: Self) -> Self
```

Comment:
I think `subtract` is a better name since reduce sounds like `map reduce` operation.